modules/GeneBurden.py

```diff
@@ -45,74 +41,94 @@ def main(
 
-    return (
+    manual_df = (
         manual_df
         # The columns practically follow the schema, only small things need to be parsed
         # 1. Confidence intervals are detached
         .withColumn(
-            'oddsRatioConfidenceIntervalLower', when(col('oddsRatio').isNotNull(), col('ConfidenceIntervalLower'))
+            'oddsRatioConfidenceIntervalLower', F.when(F.col('oddsRatio').isNotNull(), F.col('ConfidenceIntervalLower'))
         )
         .withColumn(
-            'oddsRatioConfidenceIntervalUpper', when(col('oddsRatio').isNotNull(), col('ConfidenceIntervalUpper'))
+            'oddsRatioConfidenceIntervalUpper', F.when(F.col('oddsRatio').isNotNull(), F.col('ConfidenceIntervalUpper'))
         )
-        .withColumn('betaConfidenceIntervalLower', when(col('beta').isNotNull(), col('ConfidenceIntervalLower')))
-        .withColumn('betaConfidenceIntervalUpper', when(col('beta').isNotNull(), col('ConfidenceIntervalUpper')))
+        .withColumn('betaConfidenceIntervalLower', F.when(F.col('beta').isNotNull(), F.col('ConfidenceIntervalLower')))
+        .withColumn('betaConfidenceIntervalUpper', F.when(F.col('beta').isNotNull(), F.col('ConfidenceIntervalUpper')))
         .drop('ConfidenceIntervalLower', 'ConfidenceIntervalUpper')
         # 2. Collect PMID and allelic requirements in an array
-        .withColumn('literature', array(col('literature')))
+        .withColumn('literature', F.array(F.col('literature')))
         .withColumn(
-            'allelicRequirements', when(col('allelicRequirements').isNotNull(), array(col('allelicRequirements')))
+            'allelicRequirements',
+            F.when(F.col('allelicRequirements').isNotNull(), F.array(F.col('allelicRequirements'))),
         )
         # 3. Add hardcoded values and drop URLs (will be handled by the FE) and HGNC symbols
-        .withColumn('datasourceId', lit('gene_burden'))
-        .withColumn('datatypeId', lit('genetic_association'))
+        .withColumn('datasourceId', F.lit('gene_burden'))
+        .withColumn('datatypeId', F.lit('genetic_association'))
         .drop('url', 'targetFromSource')
         .distinct()
     )
 
-
-def read_gene_burden_curation(curated_data: str, spark_instance: SparkSession) -> DataFrame:
+    # Assert that curated evidence is mapped to the correct disease
+    if (
+        manual_df.filter(
+            (
+                (F.col('projectId') == 'Autism Sequencing Consortium')
+                & (F.col('diseaseFromSourceMappedId') != 'EFO_0003756')
+            )
+            | ((F.col('projectId') == 'SCHEMA consortium') & (F.col('diseaseFromSourceMappedId') != 'MONDO_0005090'))
+            | ((F.col('projectId') == 'Epi25 collaborative') & (F.col('diseaseFromSourceMappedId') != 'MONDO_0100062'))
+        ).count()
+        != 0
+    ):
+        logging.exception('Curated evidence is not mapped to the correct disease.')
+        raise AssertionError('Curated evidence is not mapped to the correct disease.')
```

Comment:
I have a bad feeling about this... in a way you are checking the manual disease mappings at evidence level, and I'm not sure if this is good. What if EFO changes the autism spectrum disorder to MONDO_0014248 (also autism spectrum disorder) this assertion would pass, but because the mapping is outdated, the evidence would pointing to a wrong disease.

Reply:
We have agreed that the evidence generation is not the right place to do this checks. Thank you so much for the input ^^